lib/ansible/playbook/included_file.py

```diff
@@ -146,13 +148,14 @@ def process_include_results(results, iterator, loader, variable_manager):
                         if role_name is not None:
                             role_name = templar.template(role_name)
 
-                        original_task._role_name = role_name
-                        for from_arg in original_task.FROM_ARGS:
+                        original_task_copy = deepcopy(original_task)
```

Comment:
I haven't looked at this PR in depth yet, but the first thing I notice is that this should use `original_task.copy()` instead of `deepcopy`.  Our `copy` methods are specifically designed for copying these objects.

We will also need to closely monitor memory usage due to any changes that make copies of objects.

Reply:
replaced deepcopy with original_task.copy() with commit 719d27dd067dbe1194a50378a6825d6686988afa